experimental/framework/fluid-llm/src/test/shared-tree-diff/sharedTreeDiffArrays.spec.ts

```diff
@@ -0,0 +1,294 @@
+import { strict as assert } from "node:assert";
+
+import { SchemaFactory } from "@fluidframework/tree";
+
+import { sharedTreeDiff } from "../../shared-tree-diff/index.js";
+
+const schemaFactory = new SchemaFactory("TreeNodeTest");
+
+describe("sharedTreeDiff() - arrays", () => {
+	class StringArrayNode extends schemaFactory.array(
+		"StringTreeArrayNode",
+		schemaFactory.string,
+	) {}
+
+	it("top level array & array diff", () => {
+		const treeNode = new StringArrayNode(["test", "testing"]);
+		const diffs = sharedTreeDiff(treeNode as unknown as unknown[], ["test"]);
```

Comment:
Can we create an uniform type for `Record<string, unknown> | unknown[]` (e.g., `SharedTreeDiffType` or anything that makes sense)?

Reply:
I'm going to look into a strategy to avoid the type casting.  Maybe a type like you said is a good idea or maybe there is something I can do with function signatures so that things plug in nicely.